core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -703,10 +712,11 @@ public CompletionStage<Void> clearAllStores(Predicate<? super StoreConfiguration
          }
          CompletionStage<Void> stage = stageBuilder.freeze();
          if (CompletionStages.isCompletedSuccessfully(stage)) {
+            requestEnd(spanRef);
             return stage;
          } else {
             release = false;
-            return stage.whenComplete((e, throwable) -> releaseReadLock(stamp));
+            return stage.whenComplete((e, throwable) -> requestEndAndReleaseLock(stamp, spanRef, throwable));
          }
       } finally {
          if (release) {
```

Comment:
I can't link to the code directly as it is out of this view, but we need to end the span if release is true. Need to make sure to update this in all places.

Reply:
Basically what I tried to do is to bind the scope of the lock with the scope of the span.
And even if it was the correct approach, I will need another pair of eyes to check I really implemented it.

Where do we not end the span of release if true?

Thanks for the help!

Reply:
We can't use a sub class as we already have the PassivationPersistenceManager sub class, it has to use composition to do this.

Reply:
done! I'll push it in the next push :)